proof-systems/src/darlin/poly-commit/src/lib.rs

```diff
@@ -888,3 +923,20 @@ pub fn evaluate_query_map_to_vec<'a, F: Field>(
     }
     v
 }
+
+/// Transform a challenge to its representation in the field F.
+/// 'chal' bits are supposed to be in LE bit order.
+pub(crate) fn read_fe_from_challenge<F: PrimeField>(mut chal: Vec<bool>) -> Result<F, Error> {
+    
+    // Pad before reversing if necessary
+    if chal.len() < F::size_in_bits() {
+        chal.append(&mut vec![false; F::size_in_bits() - chal.len()])
+    }
+
+    // Reverse and read (as read_bits read in BE)
+    chal.reverse();
+    let scalar = F::read_bits(chal)
+        .map_err(|e| Error::Other(e.to_string()))?;
+
+    Ok(scalar)
+}
```

Comment:
I'm not entirely sure this function is really necessary.
The current implementation of `read_bits` for `F: PrimeField` discards any leading zero.
Therefore `read_fe_from_challenge` could be equivalently rewritten as
```
pub(crate) fn read_fe_from_challenge<F: PrimeField>(mut chal: Vec<bool>) -> Result<F, Error> {
    chal.reverse();
    let scalar = F::read_bits(chal).map_err(|e| Error::Other(e.to_string()))?;

    Ok(scalar)
}
```
skipping the padding. At this point we could directly interpret the bits of the challenge as a big endian sequence, and skip the `reverse()` on the challenge.
Which is to say, we could drop the function `read_fe_from_challenge` and directly use the `read_bits` on `chal` (as done in marlin).

Reply:
Of course also `challenge_to_non_native_field_element` inside `PolynomialCommitmentVerifierGadget` should be updated, dropping the call to `rev()`.

Reply:
Done